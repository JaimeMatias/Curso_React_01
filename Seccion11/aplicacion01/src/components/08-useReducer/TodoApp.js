import React, { useReducer, useEffect } from 'react'
import useForm from '../../hooks/useForm'
import './styles.css'
import { todoReducer } from './todoReducer'
import { ToDoList } from './toDoList/toDoList'
import { ToDoAdd } from './toDoAdd/toDoAdd'

const init = () => {
    return JSON.parse(localStorage.getItem('todos')) || []
}


const TodoApp = () => {
    const [todos, dispatch] = useReducer(todoReducer, [], init)


    useEffect(() => {
        localStorage.setItem('todos', JSON.stringify(todos))
    }, [todos])


    const handleDelete = (todoID) => {
        console.log(todoID)
        const action = {
            type: 'delete',
            payload: todoID
        };
        dispatch(action);
    }

    const handleToggle = (todoId) => {
        console.log(todoId)
        dispatch({
            type: 'toggle',
            payload: todoId
        })
    }

    const handleAddTodo = (newTodo) => {

        dispatch({
            type: 'add',
            payload: newTodo
        })

    }

    const handleSubmit = (e) => {



    }

    return (
        <div>
            <h1>TodoApp: {todos.length}</h1>
            <hr />

            <div className='row'>
                <ToDoList input={todos} handleDelete={handleDelete} handleToggle={handleToggle} />

                <div className='col-5'>
                    <ToDoAdd handleAddTodo={handleAddTodo} />
                </div>

            </div>







        </div>
    )
}

export default TodoApp
