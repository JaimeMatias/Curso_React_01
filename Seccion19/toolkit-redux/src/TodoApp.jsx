import React, { useState } from 'react'
import { useGetTodoQuery, useGetTodosQuery } from './store/apis/todosApi'

const TodoApp = () => {
    // const { data: todos, isLoading } = useGetTodosQuery();
    const[todoID,settodoID]=useState(1)
    const {data:todo,isLoading}= useGetTodoQuery(todoID)

    return (
        <div>
            <h1>Todos /RTK Query</h1>
            <hr />
            <h4>is Loading:{isLoading ? 'True' : 'False'}</h4>
            <pre>{JSON.stringify(todo)}</pre>
             
         {/* <ul>{todos.map(element=>( <li key={element.id}><strong>{element.completed ? 'DONE ':'Pending '}</strong>{element.title}</li>))}
       </ul>   */}
             {/* <ul>
                {todos.map(element => (
                    <li key={element.id}>
                        {element.title}
                    </li>
                ))}
            </ul>  */}
            <button on onClick={()=>{settodoID(todoID-1)}}>
                Previous Todo
            </button>
            <button on onClick={()=>{settodoID(todoID+1)}}>
                Next Todo
            </button>
        </div>
    )
}

export default TodoApp
