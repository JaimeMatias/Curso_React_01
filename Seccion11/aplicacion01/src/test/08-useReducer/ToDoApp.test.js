import React from 'react';
import { mount, shallow } from "enzyme"
import { demoTodos } from '../fixtures/demoTodos';
import { act } from '@testing-library/react-hooks';
import TodoApp from '../../components/08-useReducer/TodoApp';

describe('Pruebas en <TodoApp />>', () => {
    const wrapper= shallow(<TodoApp/>)

    Storage.prototype.setItem =jest.fn(()=>{})
  test('Debe de mostrarse correctamente ', () => {
    expect(wrapper).toMatchSnapshot()
  });
  
  test('Debe de agregar un TODO', () => {
    const wrapper = mount(<TodoApp />);
    
    act(()=>{
        wrapper.find('ToDoAdd').prop('handleAddTodo')(demoTodos[0])    
        wrapper.find('ToDoAdd').prop('handleAddTodo')(demoTodos[1])
    })
    expect(wrapper.find('h1').text().trim()).toBe('TodoApp: 2');
    expect(localStorage.setItem).toHaveBeenCalledTimes(2);
  })
  
  test('debe de eliminar un todo ', () => {
    wrapper.find('ToDoAdd').prop('handleAddTodo')(demoTodos[0])
    wrapper.find('ToDoList').prop('handleDelete')(demoTodos[0].id)
    expect(wrapper.find('h1').text().trim()).toBe('TodoApp: 0');


  })
  
})
