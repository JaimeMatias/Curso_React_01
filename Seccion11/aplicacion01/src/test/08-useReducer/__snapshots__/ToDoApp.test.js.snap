// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoApp />> Debe de mostrarse correctamente  1`] = `
<div>
  <h1>
    TodoApp: 
    0
  </h1>
  <hr />
  <div
    className="row"
  >
    <ToDoList
      handleDelete={[Function]}
      handleToggle={[Function]}
      input={Array []}
    />
    <div
      className="col-5"
    >
      <ToDoAdd
        handleAddTodo={[Function]}
      />
    </div>
  </div>
</div>
`;
